Flaky Test TestRedisCli

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.


### Version

unstable

### Minimal reproduce step

https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698

### What did you expect to see?

PASS

### What did you see instead?

```
--- FAIL: TestRedisCli (14.64s)
    --- FAIL: TestRedisCli/test_interactive_cli (0.02s)
        --- FAIL: TestRedisCli/test_interactive_cli/INFO_response_should_be_printed_raw (0.01s)
            cli_test.go:[155](https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698#step:12:156): 
                	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:155
                	Error:      	Expect "is_compac" to match "^$|^#|^[^#:]+:"
                	Test:       	TestRedisCli/test_interactive_cli/INFO_response_should_be_printed_raw
        --- FAIL: TestRedisCli/test_interactive_cli/Status_reply (0.00s)
            cli_test.go:97: 
                	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:97
                	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:[160](https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698#step:12:161)
                	Error:      	Not equal: 
                	            	expected: "OK"
                	            	actual  : "ting:no"
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1 +1 @@
                	            	-OK
                	            	+ting:no
                	Test:       	TestRedisCli/test_interactive_cli/Status_reply
        --- FAIL: TestRedisCli/test_interactive_cli/Integer_reply (0.00s)
            cli_test.go:97: 
                	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:97
                	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:[164](https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698#step:12:165)
                	Error:      	Not equal: 
                	            	expected: "(integer) 1"
                	            	actual  : "OK"
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1 +1 @@
                	            	-(integer) 1
                	            	+OK
                	Test:       	TestRedisCli/test_interactive_cli/Integer_reply
        --- FAIL: TestRedisCli/test_interactive_cli/Bulk_reply (0.00s)
            cli_test.go:97: 
                	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:97
                	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:[169](https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698#step:12:170)
                	Error:      	Not equal: 
                	            	expected: "\"foo\""
                	            	actual  : "(integer) 1"
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1 +1 @@
                	            	-"foo"
                	            	+(integer) 1
                	Test:       	TestRedisCli/test_interactive_cli/Bulk_reply
        --- FAIL: TestRedisCli/test_interactive_cli/Multi-bulk_reply (0.00s)
            cli_test.go:97: 
                	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:97
                	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:[175](https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698#step:12:176)
                	Error:      	Not equal: 
                	            	expected: "1) \"foo\"\n2) \"bar\""
                	            	actual  : "\"foo\""
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1,2 +1 @@
                	            	-1) "foo"
                	            	-2) "bar"
                	            	+"foo"
                	Test:       	TestRedisCli/test_interactive_cli/Multi-bulk_reply
        --- FAIL: TestRedisCli/test_interactive_cli/Parsing_quotes (0.00s)
            cli_test.go:97: 
                	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:97
                	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/cli/cli_test.go:[182](https://github.com/apache/incubator-kvrocks/actions/runs/3390804518/jobs/5635396698#step:12:183)
                	Error:      	Not equal: 
                	            	expected: "OK"
                	            	actual  : "1) \"foo\"\n2) \"bar\""
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1 +1,2 @@
                	            	-OK
                	            	+1) "foo"
                	            	+2) "bar"
                	Test:       	TestRedisCli/test_interactive_cli/Parsing_quotes
```

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/tests/gocase/integration/cli/cli_test.go
+++ b/tests/gocase/integration/cli/cli_test.go
@@ -84,7 +84,7 @@ func (c *interactiveCli) Read() (string, error) {
 			// to read all bytes, but it's no a good way to wait for the entire
 			// response except parsing the Redis protocol. To make this simple,
 			// we can just check whether the response has the newline or not.
-			if pos < 2 || !(b[pos-2] == '\r' && b[pos-1] == '\n') {
+			if pos < 2 || b[pos-1] != '\n' {
 				continue
 			}
 			r := string(bytes.Trim(b, "\x00"))

--- a/tests/gocase/integration/cli/cli_test.go
+++ b/tests/gocase/integration/cli/cli_test.go
@@ -23,6 +23,7 @@ import (
 	"bufio"
 	"bytes"
 	"context"
+	"errors"
 	"fmt"
 	"io"
 	"math"
@@ -67,13 +68,25 @@ func (c *interactiveCli) Close() error {
 }
 
 func (c *interactiveCli) Read() (string, error) {
+	pos := 0
 	b := make([]byte, defaultByteBufLen)
 	for {
-		n, err := c.r.Read(b)
+		if pos >= defaultByteBufLen {
+			return "", errors.New("exceed read buffer size")
+		}
+		n, err := c.r.Read(b[pos:])
 		if err != nil {
 			return "", err
 		}
+		pos += n
 		if n > 0 {
+			// For the big response size scenario, it may need multiple times
+			// to read all bytes, but it's no a good way to wait for the entire
+			// response except parsing the Redis protocol. To make this simple,
+			// we can just check whether the response has the newline or not.
+			if pos < 2 || !(b[pos-2] == '\r' && b[pos-1] == '\n') {
+				continue
+			}
 			r := string(bytes.Trim(b, "\x00"))
 			r = strings.ReplaceAll(r, "\r", "")
 			r = strings.TrimSuffix(r, "\n")

